pkg/common/config/config.go

```diff
@@ -676,3 +675,60 @@ func GetClusterFlavor(ctx context.Context) (cnstypes.CnsClusterFlavor, error) {
 	log.Error(errMsg)
 	return "", fmt.Errorf(errMsg)
 }
+
+// GetConfig loads configuration from secret and returns config object.
+func GetConfig(ctx context.Context) (*Config, error) {
+	var cfg *Config
+	var err error
+	cfgPath := GetConfigPath(ctx)
+	if cfgPath == DefaultGCConfigPath {
+		cfg, err = GetGCconfig(ctx, cfgPath)
+		if err != nil {
+			return cfg, err
```

Comment:
Should we log error before returning ? 
How can we make sure, the caller of this method will log the error ?

Reply:
printed error log before return.